lib/sentry_logging.rb

```diff
@@ -63,6 +63,13 @@ def non_nil_hash?(h)
     h.is_a?(Hash) && !h.empty?
   end
 
+  def set_raven_tag(k, v)
+    Raven.tags_context(k => v)
+    return_val = yield
+    Raven.context.tags.delete(k)
+    return_val
+  end
```

Comment:
This method is only used in your client, so I'm wondering if it makes more sense to just put in your client instead of pulling it into common code.

It expects a block, and the setting and then deleting the tag key is something that on quick inspection won't make sense without a lot of context for the _why_ -- myself included :)

So no qualms about letting you do what you need to do, but I think it makes more sense closer to the code that needs this method instead of the sentry library if at all possible.

Reply:
i set the tag and then delete it because i only want the errors inside the block to be tagged. any errors that happen afterwards shouldn't be tagged.